### PR TITLE
vstart: Remove duplicate option -N

### PR DESCRIPTION
**Pull request by kotreshhr**
Both -N and -k are essentially same, i.e to use
existing configuration without creating new conf.
So retain -N and remove -k option.

Fixes: https://tracker.ceph.com/issues/43057
Signed-off-by: Kotresh HR <khiremat@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/vstart.sh [hunk 1]
```diff
@@ -455,8 +451,22 @@ if [ $kill_all -eq 1 ]; then
     $SUDO $INIT_CEPH stop
 fi
 
-if [ "$overwrite_conf" -eq 0 ]; then
-    CEPH_ASOK_DIR=`dirname $($CEPH_BIN/ceph-conf  -c $conf_fn --show-config-value admin_socket)`
+if [ "$new" -ne 0 ]; then
```
Comment (tchaikov) [inline src/vstart.sh]: why are you transposing these two branches?

File: src/vstart.sh [hunk 2]
```diff
@@ -381,14 +385,6 @@ case $1 in
     -G)
         gssapi_authx=0
         ;;
-
-    -k )
```
Comment (tchaikov) [inline src/vstart.sh]: could you keep `-k` option instead?

File: src/vstart.sh (hunk 1 above)
Comment (kotreshhr) [reply src/vstart.sh]: I just thought of retaining the earlier check ["$new" -ne 0' ]. Fixed it.

File: src/vstart.sh (hunk 2 above)
Comment (kotreshhr) [reply src/vstart.sh]: Done

File: src/vstart.sh [hunk 3]
```diff
@@ -202,13 +202,12 @@ usage=$usage"\t-s, --standby_mds: Generate standby-replay MDS for each active\n"
...
 usage=$usage"\t-i <ip>: bind to specific ip\n"
 usage=$usage"\t-n, --new\n"
-usage=$usage"\t-N, --not-new: reuse existing cluster config (default)\n"
 usage=$usage"\t--valgrind[_{osd,mds,mon,rgw}] 'toolname args...'\n"
 usage=$usage"\t--nodaemon: use ceph-run as wrapper for mon/osd/mds\n"
 usage=$usage"\t--redirect-output: only useful with nodaemon, directs output to log file\n"
 usage=$usage"\t--smallmds: limit mds cache size\n"
 usage=$usage"\t-m ip:port\t\tspecify monitor address\n"
-usage=$usage"\t-k keep old configuration files\n"
+usage=$usage"\t-k keep old configuration files (default)\n"
```
Comment (batrick) [inline src/vstart.sh]: `default` is a change in behavior. Why?

Comment (kotreshhr) [reply src/vstart.sh]: No, It was the same behavior without this patch as well. When '-n' was not passed, '-N' was being  used which used the existing configuration files if present. In the same way, now if -n is not passed, it uses existing configuration files. Hence mentioned as default.

**Comment by kotreshhr** [comment]
Rebased

File: src/vstart.sh (hunk 3 above)
Comment (tchaikov) [reply src/vstart.sh]: yeah. before this change, "new" is 0, and "overwrite_conf" is 1 by default. but later, "overwrite_conf" is reset if  "new" is "0". so, by default, the existing conf is kept intact.

**Comment by kotreshhr** [comment]
> moreover, i think we can remove either of `new` or `overwrite_conf`. as i feel that they are also redundant.

Even I thought so initially but there is case of --inc-osd which adds osd to existing which requires distinction of 'new' and 'overwrite_conf'

File: src/vstart.sh (hunk 3 above)
Comment (tchaikov) [reply src/vstart.sh]: @batrick does this make sense with the explanations above?

**Comment by kotreshhr** [comment]
Rebased

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-12-21_03:01:01-rados-wip-kefu-testing-2019-12-21-0133-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-12-23_16:20:31-rados-wip-kefu-testing-2019-12-21-0133-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/43396
- https://tracker.ceph.com/issues/43415
- https://tracker.ceph.com/issues/43414
- https://tracker.ceph.com/issues/43416

**Comment by tchaikov** [comment]
@batrick ping?

File: src/vstart.sh (hunk 3 above)
Comment (batrick) [reply src/vstart.sh]: Yes thanks.

**Pull request merged**